ui/src/telegrafs/components/CollectorRow.tsx

```diff
@@ -159,11 +167,22 @@ class CollectorRow extends PureComponent<Props> {
     }
   }
 
+  private handleNameClick = (e: MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault()
+    this.handleOpenConfig()
+  }
+
   private handleOpenConfig = (): void => {
-    this.props.onOpenTelegrafConfig(
-      this.props.collector.id,
-      this.props.collector.name
-    )
+    const {
+      collector,
+      router,
+      org,
+      onSetTelegrafConfigID,
+      onSetTelegrafConfigName,
+    } = this.props
+    onSetTelegrafConfigID(collector.id)
+    onSetTelegrafConfigName(collector.name)
```

Comment:
How do you feel about moving these to the telegraf config overlay? So it doesnt matter where its opened from or if you were on a previous page to have this information